pyracf/resource/resource_admin.py

```diff
@@ -265,11 +265,316 @@ def set_universal_access(
         )
         return self._to_steps(result)
 
+    # ============================================================================
+    # Individual Access
+    # ============================================================================
     def get_my_access(self, resource: str, class_name: str) -> Union[str, bytes, None]:
         """Get the access associated with your own general resource profile."""
         profile = self.extract(resource, class_name, profile_only=True)
         return self._get_field(profile, "base", "yourAccess")
 
+    # ============================================================================
+    # Auditing Rules
+    # ============================================================================
+    def get_audit_rules(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes, None]:
+        """Get the auditing rules associated with this general resource profile."""
+        profile = self.extract(resource, class_name, profile_only=True)
+        return self._get_field(profile, "base", "auditing")
+
+    def clear_audit_failures(
+        self, resource: str, class_name: str
+    ) -> Union[dict, bytes]:
+        """
+        Clears the auditing rules for failures for this general resource profile while
+        preserving other rules.
+        """
+        result = [self.extract(resource, class_name)]
```

Comment:
Profile extract results should not be in the security result steps dictionary. The idea I had with the security result steps dictionary was that for any compound function, it would serve as an audit trail of all operations that make alterations. Since profile extract just retrieves information and doesn't modify anything, it should be left out.

Reply:
Ah, see I see how that design intent would be valuable, but I could also see how the steps dictionary could have value as an educational tool to understand the operations pyRACF is doing under the covers. I agree more with your stated intent, though and I will change this now.

Reply:
For the educational aspect of things here, I think the debug logging fills that role.